File: android/AppCalculadora/app/src/main/java/com/example/appcalculadora/MainActivity.java
package com.example.appcalculadora;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText valor1, valor2;
    Button btnSomar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        valor1 = (EditText)findViewById(R.id.edtValor1);
        valor2 = (EditText)findViewById(R.id.edtValor2);
        btnSomar = (Button)findViewById(R.id.btnCalcular);


    }

    public void somar(View view) {
        int v1 = Integer.parseInt(valor1.getText().toString());
        int v2 = Integer.parseInt(valor2.getText().toString());
        int resultado = v1 + v2;

        Toast.makeText(this, "A soma é: " + resultado, Toast.LENGTH_SHORT).show();
    }
}

File: android/StudentApproval/app/src/main/java/br/edu/dio/studentapproval/MainActivity.kt
package br.edu.dio.studentapproval

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val calculationButton = calculate
        val result = result

        calculationButton.setOnClickListener {
            val grade1 = Integer.parseInt(grade1.text.toString())
            val grade2 = Integer.parseInt(grade2.text.toString())
            val average = (grade1 + grade2) / 2
            val missedClasses = Integer.parseInt(missedClasses.text.toString())

            if(average >= 6 && missedClasses <= 10) {
                result.text = "Aluno aprovado com a média $average e com $missedClasses faltas!"
                result.setTextColor(Color.GREEN)
            } else {
                result.text = "Aluno reprovado com a média $average e com $missedClasses faltas!"
                result.setTextColor(Color.RED)
            }
        }
    }
}